File: app/src/main/java/home/self/beerviewer_mvvm/di/ActivityBindingModule.java
package home.self.beerviewer_mvvm.di;


import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import home.self.beerviewer_mvvm.view.beerdetail.BeerDetailActivity;
import home.self.beerviewer_mvvm.view.beerdetail.BeerDetailModule;
import home.self.beerviewer_mvvm.view.beersview.BeersViewActivity;
import home.self.beerviewer_mvvm.view.beersview.BeersViewModule;
import home.self.beerviewer_mvvm.view.splash.SplashActivity;
import home.self.beerviewer_mvvm.view.splash.SplashModule;

@Module
public abstract class ActivityBindingModule {

    @ContributesAndroidInjector(modules = SplashModule.class)
    @ActivityScope
    abstract SplashActivity splashActivity();

    @ContributesAndroidInjector(modules = BeersViewModule.class)
    @ActivityScope
    abstract BeersViewActivity beerViewActivity();

    @ContributesAndroidInjector(modules = BeerDetailModule.class)
    @ActivityScope
    abstract BeerDetailActivity beerDetailActivity();
}

File: app-kotlin/src/main/java/home/self/beerviewer_mvvm/app_kotlin/di/ActivityBindingModule.kt
package home.self.beerviewer_mvvm.app_kotlin.di


import com.mashup.dutchmarket.di.ActivityScope
import dagger.Module
import dagger.android.ContributesAndroidInjector
import home.self.beerviewer_mvvm.app_kotlin.view.beerdetail.BeerDetailActivity
import home.self.beerviewer_mvvm.app_kotlin.view.beerdetail.BeerDetailModule
import home.self.beerviewer_mvvm.app_kotlin.view.beersview.BeersViewActivity
import home.self.beerviewer_mvvm.app_kotlin.view.beersview.BeersViewModule
import home.self.beerviewer_mvvm.app_kotlin.view.splash.SplashActivity
import home.self.beerviewer_mvvm.app_kotlin.view.splash.SplashModule

@Module
internal abstract class ActivityBindingModule {

    @ContributesAndroidInjector(modules = arrayOf(SplashModule::class))
    @ActivityScope
    abstract fun splashActivity(): SplashActivity

    @ContributesAndroidInjector(modules = arrayOf(BeersViewModule::class))
    @ActivityScope
    abstract fun beersViewActivity() : BeersViewActivity

    @ContributesAndroidInjector(modules = arrayOf(BeerDetailModule::class))
    @ActivityScope
    abstract fun beerDetailActivity() : BeerDetailActivity
}
